resource/resource_registry.go

```diff
@@ -185,14 +177,21 @@ func (rs APIRegistration[ResourceT]) RegisterRPCService(
 	)
 }
 
-// AssociatedNameUpdater allows an associated config to have its names updated externally.
-type AssociatedNameUpdater interface {
+// AssociatedConfig defines the contract for a config that is associated with another config.
+type AssociatedConfig interface {
+	// Equals is a function that describes if an AssociatedConfig is the same as another
+	Equals(AssociatedConfig) bool
+
+	// UpdateResourceNames allows an AssociatedConfig to have its names updated externally.
 	UpdateResourceNames(func(n Name) Name)
+
+	// Link associates an AssociatedConfig to a specific resource model (e.g. builtin data capture).
+	Link(conf *Config)
```

Comment:
Apologies I missed that you had put `Link` in the interface already; looks good to me!